radix-engine/src/vm/wasm/prepare.rs

```diff
@@ -930,32 +789,35 @@ impl WasmModule {
     }
 
     pub fn enforce_global_limit(self, max_number_of_globals: u32) -> Result<Self, PrepareError> {
-        if let Some(section) = self.module.global_section() {
-            if section.entries().len() > max_number_of_globals as usize {
-                return Err(PrepareError::TooManyGlobals);
-            }
+        if self.module.num_local_globals() > max_number_of_globals {
+            return Err(PrepareError::TooManyGlobals);
         }
+
         Ok(self)
     }
 
     pub fn enforce_export_constraints<'a, I: Iterator<Item = &'a BlueprintDefinitionInit>>(
         self,
         blueprints: I,
     ) -> Result<Self, PrepareError> {
+        if self.module.exports_count == 0 {
+            return Err(PrepareError::NoExportSection);
+        }
```

Comment:
Why do we need to enforce this?

Reply:
Trying to keep the same approach as so [far](https://github.com/radixdlt/radixdlt-scrypto/pull/1208/files#diff-f8451cacac25f49e552023e9aaa0d43c5a7b82b5ec528059d2c5f5428af94f81L948).